project_timesheet_management/models/timesheet_invoice.py

```diff
@@ -0,0 +1,764 @@
+                    timesheet_invoice.project_id.resource_ids.mapped(
+                        'user_ids').filtered(lambda user: user.has_group(po_group))
+                mail_state = 'Uploaded'
+                for user_id in user_ids:
+                    mail_template.write({
+                        'email_to': user_id.email or '',
+                        'email_from': self.env.user.email or '',
+                        'reply_to': user_id.email or ''})
+                    mail_template.sudo().with_context(state=mail_state).send_mail(
+                        timesheet_invoice.id, force_send=True)
+        return res
+
+    def button_rejected(self):
+        """
+        Set timesheet invoice to Reject State.
+        Updating with sudo env due to the access rights.
+        """
+        mail_template = self.env.ref(
+            "project_timesheet_management.email_timesheet_invoice_confirmed"
+        )
+
+        po_group = 'project_management_security.group_supp_project_coordinator'
+        cann_po_group = 'project_management_security.group_project_coordinator'
+        # )
+        author_id = (
+            self.env["res.users"].sudo().browse(request.session.uid).partner_id.id
+        )
+        for timesheet_inv in self:
+            u_lst = []
+            mail_status = 'Rejected'
+            if timesheet_inv.state == "pre-approved":
+                mail_template.write({
+                    'email_from': self.env.user.email or self.env.user.partner_id.email,
+                    'email_to': timesheet_inv.project_id.user_id.email or '',
+                    'reply_to': self.env.user.email or self.env.user.partner_id.email})
+                mail_template.sudo().with_context(state=mail_status).send_mail(
+                    timesheet_inv.id, force_send=True,
+                    email_values={"author_id": author_id or False})
+                u_lst = [timesheet_inv.project_id.user_id.id]
+            elif timesheet_inv.state == "confirm":
+                mail_template.write({
+                    'email_from': self.env.user.email or self.env.user.partner_id.email,
+                    'email_to': timesheet_inv.pre_approved_by.email or '',
+                    'reply_to': self.env.user.email or self.env.user.partner_id.email})
+                mail_template.sudo().with_context(state=mail_status).send_mail(
+                    timesheet_inv.id, force_send=True,
+                    email_values={"author_id": author_id or False})
+                u_lst = [timesheet_inv.pre_approved_by.id]
+            timesheet_inv.with_context(from_approval=True).sudo().write(
+                {"state": "rejected"}
+            )
+
+            # Project Manager
+            if (
+                timesheet_inv.project_id.user_id.id not in u_lst
+                and timesheet_inv.project_id.product_owner_id.id == self.env.user.id
+            ):
+                mail_template.write({
```

Comment:
You should not write to the email template, instead you can use email_values as parameter for send_mail function and no need to define reply_to  separately as it uses the same value as email_from